src/components/plots/GeneSearchBar.jsx

```diff
@@ -25,31 +26,56 @@ const GeneSearchBar = (props) => {
 
   const [options, setOptions] = useState([]);
 
-  // pass reactive component as value (search text) to allow auto clear on select
   const [value, setValue] = useState('');
 
   const onOptionSelect = (newGene) => {
-    const genes = [...config?.selectedGenes, newGene];
-
-    onSelect(genes);
-    setValue('');
+    const genes = value.split(/(?<!-)[,\s]+(?!-)/);
+    genes.splice(-1, 1, `${newGene}, `);
+    setValue(genes.join(', '));
+    setOptions([]);
   };
 
-  const onSearch = (searchText) => {
-    setValue(searchText);
+  const onSearch = (input) => {
+    setValue(input);
+
+    const genes = input.split(/(?<!-)[,\s]+(?!-)/);
```

Comment:
This regex pattern is used 3 times in this file. It's not good practice to have duplicated code. Can you refactor it into a variable to avoid duplication?

Reply:
I moved the regex outside the functions to avoid duplications, where it becomes `const genes = value.split(/(?<!-)[,\s]+(?!-)/);` but I'm running into a problem with `onSearch` where i both update `value` and need to use the updated `value` to define `searchText` for filtering options. I think i need to duplicate the regex there, because it needs to use `input` instead of `value`.

Reply:
Oh no, I mean take out the Regex. Can you do something like

    const GENES_REGEX = /(?<!-)[,\s]+(?!-)/
    ....
   const genes = values.split(GENES_REGEX)
    ....
    const inputGenes = input.split(GENES_REGEX)